contracts/cw1155-base/src/state.rs

```diff
@@ -0,0 +1,23 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{CanonicalAddr, Uint128};
+use cw1155::Expiration;
+use cw_storage_plus::{Item, Map};
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct Approval {
+    /// Account that can transfer/send the token
+    pub spender: CanonicalAddr,
+    /// When the Approval expires (maybe Expiration::never)
+    pub expires: Expiration,
+}
+/// Store the minter address who have permission to mint new tokens.
```

Comment:
Nice job with the state, and good to comment each one what those &[u8] and &str mean :+1: